woodwork/tests/logical_types/test_logical_types.py

```diff
@@ -40,3 +52,24 @@ def test_ordinal_init_with_order():
     order = ('bronze', 'silver', 'gold')
     ordinal_from_tuple = Ordinal(order=order)
     assert ordinal_from_tuple.order == order
+
+
+@pytest.mark.parametrize(**COLUMN_TO_LOGICAL_TYPE)
+def test_init(sample_df, column, logical_type):
+    sample_df.ww.init(logical_types={column: logical_type})
+    actual = sample_df.ww[column].ww.logical_type
+    info = f'"{column}" not initialized as "{logical_type}"'
+    assert actual == logical_type, info
+
+
+@pytest.mark.parametrize(**COLUMN_TO_LOGICAL_TYPE)
+def test_set_types(sample_df, column, logical_type):
+    sample_df.ww.init()
+    before = sample_df.ww[column].ww.logical_type
+    info = f'"{column}" already set as "{logical_type}""'
+    assert before != logical_type, info
+
+    sample_df.ww.set_types(logical_types={column: logical_type})
+    after = sample_df.ww[column].ww.logical_type
+    info = f'"{column}" not set as "{logical_type}"'
+    assert after == logical_type, info
```

Comment:
I'm thinking we do not need these specific tests here. Instead we should update our tests such as `test_sets_int64_dtype_on_init()` to include these new types in the list of logical types that get used. Additionally, we should review our mutual information and describe tests to make sure these types are includes, as they should be valid columns as input for those functions.

Reply:
Are there existing tests that check if columns can be set to logical types using `set_types` for all logical types?

Reply:
No we don't check setting the logical types after init for all logical types. However, the key steps involved in setting a type after init are the same as those during init, namely parsing the logical type with the `_parse_logical_type` function and updating the column dtype with the `_update_column_dtype` function. 

Based on that I'm thinking it is not necessary to test all logical types in both places (init and set_types) unless there is something unique here that isn't covered by the other tests.

Reply:
I can see how those tests are arguably not needed. Let's review with @gsheni since he asked to test setting this logical type (https://github.com/alteryx/woodwork/pull/849#issuecomment-825322199)

Reply:
@jeff-hernandez I was mistaken and @thehomebrewnerd is correct. 

Let's update `test_sets_int64_dtype_on_init` to include this new types. Mutual Info should work with Age and AgeNullable. Describe should work as well.

Reply:
@gsheni okay, will update

Reply:
@thehomebrewnerd which mutual info and describe tests do we want to include these types in?

Reply:
For describe, I would update `test_describe_accessor_method` to include `AgeNullable` and `Age` in the lists described here: https://github.com/alteryx/woodwork/blob/ce17d7399c77d75f82930c514ea91dccfa980f5d/woodwork/tests/accessor/test_statistics.py#L240-L241

Adding them to the list should be all you need to do, and will cause them to get tested below in the appropriate sections of this test where those lists get used.

Reply:
As for mutual info, I'm thinking the update you already made to confirm `Age` and `AgeNullable` are returned by `get_valid_mi_types` should be sufficient. I'm not thinking there is additional value in testing a mutual information calculation with those types actually used.

I think we also need to update the docstrings for the accessor `mutual_information` and `mutual_information_dict` methods to include these logical types (and the nullable integer and boolean types, which we neglected to add before).